pkgs/by-name/em/emocli/package.nix

```diff
@@ -0,0 +1,32 @@
+{ lib, nimPackages, musl, fetchurl, fetchFromGitLab }:
+
+nimPackages.buildNimPackage rec {
+  pname = "emocli";
+  version = "1.0.0";
+  src = fetchFromGitLab {
+    owner = "AsbjornOlling";
+    repo = "emocli";
+    rev = "v${version}";
+    hash = "sha256-yJu+8P446gzRFOi9/+TcN8AKL0jKHUxhOvi/HXNWL1A=";
+  };
+  nativeBuildInputs = [ musl ];
+  nimFlags = [
+    "-d:release"
+    "--gcc.exe:musl-gcc"
+    "--gcc.linkerexe:musl-gcc"
+    "--passL:-static"
+    "--maxLoopIterationsVM:1000000000"
+  ];
+  doCheck = true;
+  EMOCLI_DATAFILE = fetchurl {
+    url = "https://unicode.org/Public/emoji/15.1/emoji-test.txt";
```

Comment:
Also, I think you might have a licensing issue, as I can't see to find the below license anywhere in your project?

[https://www.unicode.org/license.txt](https://www.unicode.org/license.txt)

I suggest that this gets added to the nix package.

Reply:
This has left me a bit confused, since emocli doesn't ship with the unicode data files.
It reads the data files at compile-time, in order to obtain all the emoji symbols.

The resulting binary *does* include unicode characters, but surely it's not required to include the unicode license whenever I use a unicode character?
If that's the case, it becomes a hassle to spell my name, or the city I live in.

:man_shrugging: but including the license shouldn't be a problem anyway, regardless of whether it's necessary or not.

I couldn't find a way to add multiple licenses to the `meta` section of the emocli package, so I figured that I'd have to make the unicode data files a separate package.
Luckily, there already is the `unicode-emoji` package in nixpkgs! That package correctly lists the unicode license in the `meta` section. Using the existing package is cleaner anyways.

There is the small issue of nixpkg's `unicode-emoji` being at an older version, so it's missing a few emoji compared to before. But I'll just make a separate PR to update the `unicode-emoji` package, once this one is merged :slightly_smiling_face:

Reply:
I've pushed a new verison of the package, that now reads the unicode data files from the `unicode-emoji` nix packge, instead of getting them using fetchurl. Hopefully that resolves any license problems.

Reply:
You're correct about you don't have to include the license whenever you use a Unicode character. Without being an expert on licenses. Then as I read it the problem is that you distribute software containing the Unicode emojis but if there already is a package for the emoji's it would be better to use that and let that package handle the licensing. Sorry for not noticing it before 😊

Reply:
@MikaelFangel 
No worries at all. I wouldn't expect you to read the source code of my project :)

Thanks for being patient with me, and thanks for reviewing my nix code!